### PR TITLE
Fix/ccip 2466 homechain config sync

### PR DESCRIPTION
**Pull request by asoliman92**
Fix some of previous PR commit as it was merged with auto-merge before addressing all of them.

### CODE REVIEW

File: core/services/ocr3/plugins/ccip/internal/reader/home_chain_test.go [hunk 1]
```diff
@@ -200,9 +130,11 @@ func Test_PollingWorking(t *testing.T) {
...
 
 	ctx := context.Background()
-	_ = configPoller.Start(ctx)
-	_ = configPoller.Close()
-	time.Sleep(100 * time.Millisecond)
+	err := configPoller.Start(ctx)
+	assert.NoError(t, err)
+	err = configPoller.Close()
+	assert.NoError(t, err)
+	time.Sleep(2 * time.Second)
```
Comment (winder) [inline core/services/ocr3/plugins/ccip/internal/reader/home_chain_test.go]: Sorry, but if I'm reading the code correctly this still doesn't test the polling logic. You need to assert that `homeChainReader` is called twice. Once when `Start` is called, and once when `poll` is called.

When you call `Close()` the stop channel prevents poll from fetching a result so you never hit that block of code.

I haven't used StopChan before so I wrote this test to verify the expected behavior:
```
func TestStopChan(t *testing.T) {
	ch := make(StopChan)
	ctx, _ := ch.NewCtx()
	go func() {
		time.Sleep(1 * time.Second) // 1 second timeout
		close(ch)
	}()

	ticker := time.NewTicker(400 * time.Millisecond) // 2 ticks before the stop chan is triggered
	for {
		select {
		case <-ctx.Done():
			fmt.Println("We're done here.")
			return
		case <-ticker.C:
			fmt.Println("Ticking here.")
		}
	}
}
```

**Review by 0xnogo — APPROVED** [review]

**Review by 0xnogo — APPROVED** [review]

File: core/services/ocr3/plugins/ccip/internal/reader/home_chain_test.go (hunk 1 above)
Comment (winder) [reply core/services/ocr3/plugins/ccip/internal/reader/home_chain_test.go]: StopChan also seems totally unnecessary:
```
func TestStopChan(t *testing.T) {
	stopCh := make(chan struct{})
	go func() {
		time.Sleep(1 * time.Second)
		close(stopCh)
	}()

	ticker := time.NewTicker(500 * time.Millisecond)
	for {
		select {
		case <-stopCh:
			fmt.Println("We're done here.")
			return
		case <-ticker.C:
			fmt.Println("Ticking here.")
		}
	}
}
```

**Review by winder — CHANGES_REQUESTED** [review]
I think the polling test is still wrong

File: core/services/ocr3/plugins/ccip/internal/reader/home_chain_test.go (hunk 1 above)
Comment (asoliman92) [reply core/services/ocr3/plugins/ccip/internal/reader/home_chain_test.go]: Got you, I used the stop channel as it's used in some other polling services we have in our repos. I fixed the test to check for the number of calls now and it's working correctly. Thanks!

**Review by dimkouv — APPROVED** [review]

**Review by winder — APPROVED** [review]

**Pull request merged**